lib/consensus/raft/config.go

```diff
@@ -0,0 +1,143 @@
+package consensus
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"time"
+
+	hraft "github.com/hashicorp/raft"
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/lotus/node/config"
+	"github.com/filecoin-project/lotus/node/repo"
+)
+
+// Configuration defaults
+var (
+	DefaultDataSubFolder        = "raft-cluster"
+	DefaultWaitForLeaderTimeout = 15 * time.Second
+	DefaultCommitRetries        = 1
+	DefaultNetworkTimeout       = 100 * time.Second
+	DefaultCommitRetryDelay     = 200 * time.Millisecond
+	DefaultBackupsRotate        = 6
+	DefaultDatastoreNamespace   = "/r"
+)
+
+// ClusterRaftConfig allows to configure the Raft Consensus component for the node cluster.
+type ClusterRaftConfig struct {
+	// config to enabled node cluster with raft consensus
+	ClusterModeEnabled bool
+	// will shutdown libp2p host on shutdown. Useful for testing
+	HostShutdown bool
+	// A folder to store Raft's data.
+	DataFolder string
+	// InitPeerset provides the list of initial cluster peers for new Raft
+	// peers (with no prior state). It is ignored when Raft was already
+	// initialized or when starting in staging mode.
+	InitPeerset []string
+	// LeaderTimeout specifies how long to wait for a leader before
+	// failing an operation.
+	WaitForLeaderTimeout time.Duration
+	// NetworkTimeout specifies how long before a Raft network
+	// operation is timed out
+	NetworkTimeout time.Duration
+	// CommitRetries specifies how many times we retry a failed commit until
+	// we give up.
+	CommitRetries int
+	// How long to wait between retries
+	CommitRetryDelay time.Duration
+	// BackupsRotate specifies the maximum number of Raft's DataFolder
+	// copies that we keep as backups (renaming) after cleanup.
+	BackupsRotate int
+	// Namespace to use when writing keys to the datastore
+	DatastoreNamespace string
```

Comment:
I don't think we use this namespace anywhere

Reply:
seems like its used the in OfflineState function

Reply:
Yeah, but that's commented out, and even if it wasn't datastore namespaces aren't something that should be exposed in the user config